File: GeeksForGeeks-Problems/JAVASCRIPT/Functional Programming/Find One Extra Character.js
// User function Template for javascript

class Solution {
  extraChar(s1, s2) {
    // write your code here
    let mp = new Map();
    for (let i = 0; i < s1.length; i++) {
      mp.set(s1[i], (mp.get(s1[i]) || 0) + 1);
    }
    for (let i = 0; i < s2.length; i++) {
      mp.set(s2[i], (mp.get(s2[i]) || 0) - 1);
    }
    for (let [char, count] of mp) {
      if (count != 0) {
        return char;
      }
    }
  }
}

File: GeeksForGeeks-Problems/CPP/Miscellaneous/Find One Extra Character.cpp
// User function Template for C++

char extraChar(string &s1, string &s2)
{
    unordered_map<char, int> mp;
    for (char c : s1)
        mp[c]++;
    for (char c : s2)
        mp[c]--;
    for (auto i : mp)
    {
        if (i.second != 0)
        {
            return i.first;
        }
    }
}